index_test.go

```diff
@@ -349,6 +349,23 @@ func TestIndex_Find(t *testing.T) {
 		assert.Equal(t, 1, count)
 	})
 
+	t.Run("ok - not nil", func(t *testing.T) {
+		q := New(NotNil(key))
+		count := 0
+
+		err := db.View(func(tx *bbolt.Tx) error {
+			b := testBucket(t, tx)
+			return i.Iterate(b, q, func(key Reference, value []byte) error {
+				count++
+				return nil
+			})
+		})
+
+		assert.NoError(t, err)
+		// it's a triple index where 4 matching trees exist
+		assert.Equal(t, 4, count)
```

Comment:
I don't understand this, in `jsonExample` and `jsonExample2` there's summed up only 2 `part` fields?

And there's no non-matching document in the store? (testing `NotNil` doesn't match nil fields?)

Reply:
undoubling is done at a different level (QueryPlan collector)